lib/belay_brokerage.ex

```diff
@@ -35,6 +39,35 @@ defmodule BelayBrokerage do
     Repo.get(Investor, investor_id, prefix: partner_id)
   end
 
+  @spec holding_transaction(String.t(), String.t(), String.t(), Decimal.t()) ::
+          {:ok, Holding.t()} | {:error, Ecto.Changeset.t()}
+  def holding_transaction(partner_id, investor_id, sym, qty_delta) do
+    :ok = Transactions.publish_transaction(investor_id, sym, qty_delta)
+
+    case Repo.get_by(Holding, [investor_id: investor_id, sym: sym], prefix: partner_id) do
```

Comment:
Add a Repo.transaction here and publish message only after a successful insert,update,delete

Reply:
Done